modules/opengl/src/shader/shader.cpp

```diff
@@ -169,8 +129,8 @@ Shader::Shader(const char *vertexFilename, const char *geometryFilename,
 
 Shader::Shader(const Shader &rhs) : program_{rhs.program_}, warningLevel_{rhs.warningLevel_} {
     for (auto &elem : rhs.shaderObjects_) {
-        shaderObjects_.emplace(
-            elem.first, ShaderAttachment(this, std::make_unique<ShaderObject>(elem.second.obj())));
+        shaderObjects_.emplace_back(elem);
+        attached_.emplace_back(false);
```

Comment:
what about the callbacks here? Isn't the invariant here that `shaderObjects_`, `attached_`, and `callbacks_` have the same lengths?

Reply:
Yes, think I had changes that, but seems I lost some changes...